simplify all ReachabilyPrograms instead of just base

arifogel opened the pull request.

dhalperi commented:
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[projects/batfish/src/main/java/org/batfish/z3/ReachabilityProgram.java, line 71 at r2](https://reviewable.io:443/reviews/batfish/batfish/977#-L607rauOYWflkowmbbh:-L607rauOYWflkowmbbi:b65j589) ([raw file](https://github.com/batfish/batfish/blob/149adf7a047b3f2bc995855bce84b2e15e707fc7/projects/batfish/src/main/java/org/batfish/z3/ReachabilityProgram.java#L71)):*
> ```Java
>                 .map(s -> (RuleStatement) s)
>                 .collect(ImmutableList.toImmutableList())
>             : rules;
> ```

did you not still want the `ImmutableList.copyOf(rules)` for the false case?

---

*[projects/batfish/src/main/java/org/batfish/z3/Synthesizer.java, line 126 at r2](https://reviewable.io:443/reviews/batfish/batfish/977#-L609atUO09PWwu7BNMf:-L609atUO09PWwu7BNMg:b2ioyeg) ([raw file](https://github.com/batfish/batfish/blob/149adf7a047b3f2bc995855bce84b2e15e707fc7/projects/batfish/src/main/java/org/batfish/z3/Synthesizer.java#L126)):*
> ```Java
>     builder.setRules(rawRuleStatements);
>     builder.setInput(_input);
>     return builder.build();
> ```

minor, but if you put the filter statement first, you might end up with cleaner code:

```java
List<RuleStatement> rawRuleStatements = ...;
return ReachabilityProgram.builder().setInput().setRules().build();
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/977)*
<!-- Sent from Reviewable.io -->

arifogel commented:
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[projects/batfish/src/main/java/org/batfish/z3/ReachabilityProgram.java, line 71 at r2](https://beta.reviewable.io/reviews/batfish/batfish/977#-L607rauOYWflkowmbbh:-L62rf2A84Uo6DPPXjKX:b-886gsx) ([raw file](https://github.com/batfish/batfish/blob/149adf7a047b3f2bc995855bce84b2e15e707fc7/projects/batfish/src/main/java/org/batfish/z3/ReachabilityProgram.java#L71)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

did you not still want the `ImmutableList.copyOf(rules)` for the false case?
</blockquote></details>

Fixed

---

*[projects/batfish/src/main/java/org/batfish/z3/Synthesizer.java, line 126 at r2](https://beta.reviewable.io/reviews/batfish/batfish/977#-L609atUO09PWwu7BNMf:-L62rgEr8sdAYAwE1xEL:br1yl2r) ([raw file](https://github.com/batfish/batfish/blob/149adf7a047b3f2bc995855bce84b2e15e707fc7/projects/batfish/src/main/java/org/batfish/z3/Synthesizer.java#L126)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

minor, but if you put the filter statement first, you might end up with cleaner code:

```java
List<RuleStatement> rawRuleStatements = ...;
return ReachabilityProgram.builder().setInput().setRules().build();
```
</blockquote></details>

Simplified the whole thing

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/batfish/batfish/977)*
<!-- Sent from Reviewable.io -->

dhalperi commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/977#-:-L62t51kjqJggro0D-TW:bnfp4nl)*
<!-- Sent from Reviewable.io -->
